src/logworker.cpp

```diff
@@ -111,15 +113,38 @@ namespace g3 {
    }
 
    void LogWorker::addWrappedSink(std::shared_ptr<g3::internal::SinkWrapper> sink) {
-      auto bg_addsink_call = [this, sink] {_impl._sinks.push_back(sink);};
+
+      // capturing sink by reference is safe because this method waits for the completion of this task
+      // passing sink by value may lead to keeping it in other thread's context even after this method returns
```

Comment:
You just proved that it wasn't safe. That it could be kept alive longer. 

I.e reference to shared_ptr is a not OK